endpoint/http/src/main/java/de/fraunhofer/iosb/ilt/faaast/service/endpoint/http/request/mapper/SetSubmodelElementValueByPathRequestMapper.java

```diff
@@ -33,31 +35,41 @@
 import io.adminshell.aas.v3.model.Submodel;
 import io.adminshell.aas.v3.model.SubmodelElement;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 
 
 /**
  * class to map HTTP-PUT-Request path:
  * submodels/{submodelIdentifier}/submodel/submodel-elements/{idShortPath}
+ * <br>
+ * shells/{aasIdentifier}/aas/submodels/{submodelIdentifier}/submodel/submodel-elements/{idShortPath}
  */
-public class SetSubmodelElementValueByPathRequestMapper extends RequestMapper {
+public class SetSubmodelElementValueByPathRequestMapper extends SubmodelInterfaceRequestMapper<SetSubmodelElementValueByPathRequest<?>, SetSubmodelElementValueByPathResponse> {
 
-    private static final HttpMethod HTTP_METHOD = HttpMethod.PUT;
-    private static final String PATTERN = "^submodels/(.*?)/submodel/submodel-elements/(.*)$";
-    private static final String QUERYPARAM1 = "content";
-    private static final String QUERYVALUE1 = "value";
+    private static final String SUBMODEL_ELEMENT_PATH = "submodelElementPath";
+    private static final String PATTERN = String.format("submodel-elements/(?<%s>.*)", SUBMODEL_ELEMENT_PATH);
+    private static final String QUERY_PARAMETER_CONTENT = "content";
+    private static final String QUERY_PARAMETER_CONTENT_VALUE = "value";
 
     public SetSubmodelElementValueByPathRequestMapper(ServiceContext serviceContext) {
-        super(serviceContext);
+        super(serviceContext, HttpMethod.PUT, PATTERN);
     }
 
 
     @Override
-    public Request parse(HttpRequest httpRequest) {
-        final List<Key> path = ElementPathHelper.toKeys(EncodingHelper.urlDecode(httpRequest.getPathElements().get(4)));
-        final Identifier identifier = IdentifierHelper.parseIdentifier(EncodingHelper.base64Decode(httpRequest.getPathElements().get(1)));
+    public boolean matches(HttpRequest httpRequest) {
+        return super.matches(httpRequest)
+                && httpRequest.hasQueryParameter(QUERY_PARAMETER_CONTENT)
+                && Objects.equals(httpRequest.getQueryParameter(QUERY_PARAMETER_CONTENT), QUERY_PARAMETER_CONTENT_VALUE);
+    }
+
+
+    @Override
+    public SetSubmodelElementValueByPathRequest doParse(HttpRequest httpRequest, Map<String, String> urlParameters, OutputModifier outputModifier) throws InvalidRequestException {
+        final List<Key> path = ElementPathHelper.toKeys(EncodingHelper.urlDecode(urlParameters.get(SUBMODEL_ELEMENT_PATH)));
+        final Identifier identifier = IdentifierHelper.parseIdentifier(EncodingHelper.base64Decode(urlParameters.get(SUBMODEL_ID)));
         return SetSubmodelElementValueByPathRequest.builder()
-                .id(identifier)
```

Comment:
Add `submodelId` here?

Reply:
This happens in parent class https://github.com/FraunhoferIOSB/FAAAST-Service/blob/139197116ce904bf9245fba390fa2328221ace0d/endpoint/http/src/main/java/de/fraunhofer/iosb/ilt/faaast/service/endpoint/http/request/mapper/SubmodelInterfaceRequestMapper.java#L135